Added new receivers to action group document.

shriku11 opened the pull request:
This checklist is used to make sure that common issues in a pull request are addressed. This will expedite the process of getting your pull request merged and avoid extra work on your part to fix issues discovered during the review process. 

### PR information
- [ ] The title of the PR is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).
- [ ] Except for special cases involving multiple contributors, the PR is started from a fork of the main repository, not a branch.
- [ ] If applicable, the PR references the bug/issue that it fixes.
- [ ] Swagger files are correctly named (e.g. the `api-version` in the path should match the `api-version` in the spec).

### Quality of Swagger
- [ ] I have read the [contribution guidelines](https://github.com/Azure/azure-rest-api-specs/blob/master/.github/CONTRIBUTING.md).
- [ ] My spec meets the review criteria:
  - [ ] The spec conforms to the [Swagger 2.0 specification](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md).
  - [ ] The spec follows the guidelines described in the [Swagger checklist](../documentation/swagger-checklist.md) document.
  - [ ] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.

Commit:
Added new receivers to action group document.

Commit:
Added new receivers to action group document.

fearthecowboy commented:
Trading with @marstr  for a data-plane one (this looks pretty simple, so ... good trade :D )

marstr commented on specification/monitor/resource-manager/microsoft.insights/2017-04-01/actionGroups_API.json:
```diff
@@ -391,6 +391,20 @@
+                    "items": {
+                        "$ref": "#/definitions/ItsmReceiver"
+                    },
+                    "description": "The list of Itsm receivers that are part of this action group."
```
[nit] Because "Itsm" looks so conspicuously like "Items", in the description I would either write it out or at least have it capitalized to "ITSM" to demonstrate that it is an acronym.

marstr commented on specification/monitor/resource-manager/microsoft.insights/2017-04-01/actionGroups_API.json:
```diff
@@ -459,6 +473,55 @@
                     "description": "The URI where webhooks should be sent."
                 }
             },
+            "ItsmReceiver": {
```
I believe this is the root cause of [the failures from our CI system](https://travis-ci.org/Azure/azure-rest-api-specs/jobs/310273380#L725). I think there's a curly-brace missing above this line.

marstr reviewed (changes requested):
There are some semantic errors that need to be fixed up before we go too far. :)

Commit:
fixed few comments.

Commit:
fixed few comments.

Commit:
Corrected the file syntactically using openapi-linter

Commit:
Corrected the file syntactically using openapi-linter

marstr commented on specification/monitor/resource-manager/microsoft.insights/2017-04-01/actionGroups_API.json:
```diff
@@ -464,6 +478,55 @@
                 "serviceUri"
             ]
         },
+        "ItsmReceiver": {
```
Because these are not required properties, I don't believe this is a breaking change.

ravbhatnagar replied:
Adding new properties - required or optional is considered a breaking change. This could result in clients unitentionally overriding the properties on the server. Imagine a resource got created with this new property. Then the GET on the resource happened from a client which did not understand this property and so ignored it. Now if the PUT happens from the same client using the resource that was obtained via GET, the value of this property on the server will be overridden.

ravbhatnagar replied:
So it is advisable to increment the api-version for this.

shriku11 replied:
This is already released for this API version and is almost like a baseline now. In future we are planning to support patch operation for individual receiver types. We will change the version then.

marstr reviewed (changes requested):
There are couple of outstanding Litner errors that were introduced which I'd like to see cleaned up (find them [here](https://travis-ci.org/Azure/azure-rest-api-specs/jobs/311578946#L730)). Other than that, I think this PR is good to go :)

marstr commented:
Too be clear, I'm only asking to fix the two new linter errors that are introduced with this change, not all six.

shriku11 commented:
Ping

Commit:
Add required patch operation for which it is flagging this documentation

Commit:
Add required patch operation for which it is flagging this documentation

Commit:
Fixed some liter format validation issues, added patch example

Commit:
Fixed some liter format validation issues, added patch example

marstr reviewed (dismissed):
Thanks for making those updates, things are definitely coming together. There are still some out-lying CI errors to take care of:
https://travis-ci.org/Azure/azure-rest-api-specs/jobs/315048599#L693

In particular, it seems like there are some properties in the examples that aren't present in the swagger. Either removing them from the examples or adding them into the Swagger is pretty important.

Commit:
Fixed couple for validation errors

Commit:
Fixed couple for validation errors

Commit:
Fixed new validation error for runbook receiver

Commit:
Fixed new validation error for runbook receiver

ravbhatnagar reviewed (commented):
Please take a look at comments below

shriku11 commented on specification/monitor/resource-manager/microsoft.insights/2017-04-01/actionGroups_API.json:
```diff
@@ -464,6 +531,90 @@
+                    "type": "string",
+                    "description": "Unique identification of ITSM connection among multiple defined in above workspace."
+                },
+                "ticketConfiguration": {
```
It is JSON configuration string.

shriku11 replied:
Will fix the description.

shriku11 replied:
Good catch, the description is fixed in new iteration.

shriku11 commented on specification/monitor/resource-manager/microsoft.insights/2017-04-01/actionGroups_API.json:
```diff
@@ -464,6 +531,90 @@
+                    "type": "string",
+                    "description": "XML blob for the configurations of the ITSM action. CreateMultipleWorkItems option will be part of this blob as well."
+                },
+                "region": {
```
You mean region field? It is already released as string field for better readability.

shriku11 commented on specification/monitor/resource-manager/microsoft.insights/2017-04-01/actionGroups_API.json:
```diff
@@ -464,6 +531,90 @@
+                    "type": "boolean",
+                    "description": "Indicates whether this instance is global runbook."
+                },
+                "name": {
```
This can be used as friendly name by user also webhookResourceId format is different with subscriptionid resourcegroup details etc.

Commit:
fixed the description field for ITSM Ticket configuration field.

Commit:
fixed the description field for ITSM Ticket configuration field.

shriku11 commented:
@marstr, please resolve the conflicts by over writing old files by new ones. I do not have access to resolve the conflicts.

marstr commented:
> @shriku11 says:
@marstr, please resolve the conflicts by over writing old files by new ones. I do not have access to resolve the conflicts.

I'm not exactly sure what you mean about not having access, we should work out whatever troubles you were having. This PR is based on your fork, so there shouldn't be anything you don't have access to.

That said, I have fixed the merge conflicts for you. :)

Commit:
merge from remote

Commit:
Added example file for patch operation to new directory called stable.

shriku11 commented:
@marstr,
I got the merge conflict button disabled so thought I do not have rights to resolve it.
I also see you have changed the branch from current to master for for this PR. That resulted in conflicts as there was directory called Stable in master and not in Current branch.
Also while resolving conflicts looks like unlike other files you forgot to add the patchActionGroup.json in the directory.
I have added that and now getting following error,
FATAL: swagger-document/loader - FAILED
FATAL: Error: Could not read 'file:///git-restapi/specification/monitor/resource-manager/microsoft.insights/stable/2017-05-01-preview/diagnosticsSettings_API.json'.

Any idea?

Thanks,
-Shripad
--------------------------------------------
On Wed, 1/3/18, Martin Strobel <notifications@github.com> wrote:

 Subject: Re: [Azure/azure-rest-api-specs] Added new receivers to action group document. (#2077)
 To: "Azure/azure-rest-api-specs" <azure-rest-api-specs@noreply.github.com>
 Cc: "shriku11" <shriku1@yahoo.com>, "Mention" <mention@noreply.github.com>
 Date: Wednesday, January 3, 2018, 9:20 AM
 
 
 @shriku11 says:
 
 @marstr, please
 resolve the conflicts by over writing old files by new ones.
 I do not have access to resolve the conflicts.
 
 I'm not exactly sure what you mean about not having
 access, we should work out whatever troubles you were
 having. This PR is based on your fork, so there
 shouldn't be anything you don't have access to.
 That said, I have fixed the merge conflicts for you.
 :)
 
 —
 You are receiving this because you were mentioned.
 Reply to this email directly, view
 it on GitHub, or mute
 the thread.

marstr commented:
@ravbhatnagar would you mind taking a look at the most recent changes to see if your comments have been addressed?

marstr commented:
I'm needing to load-balance some of my PRs. I'm going to reassign this to @veronicagg.

ravbhatnagar commented:
@shriku11 - Signing off on this. But we need to make sure going forward the APIs are reviewed and approved before the changes are deployed to service side. The Microsoft REST API versioning guidance should be followed when shipping changes to the service.

veronicagg commented:
@ravbhatnagar thanks for approving from ARM side. 
@shriku11 the issue is from the path in the readme file. I created and merged a PR to fix it (since it's anyway independent from this change) - https://github.com/Azure/azure-rest-api-specs/pull/2227. Could you update this PR by pulling from master? CI should kick off again.

Commit:
Merge branch 'master' into documentation

shriku11 commented:
Merged from master after fix from @veronicagg

veronicagg reviewed (approved):
PR looks good to me. 
Labeling with potential-sdk-breaking-changes, as there are new properties added on existing api-version.

AutorestCI commented:
[![Build Status](https://travis-ci.org/AutorestCI/azure-sdk-for-ruby.svg?branch=RestAPI-PR2077)](https://travis-ci.org/AutorestCI/azure-sdk-for-ruby) https://github.com/Azure/azure-sdk-for-ruby/pull/1227

AutorestCI commented:
[![Build Status](https://travis-ci.org/AutorestCI/azure-sdk-for-python.svg?branch=RestAPI-PR2077)](https://travis-ci.org/AutorestCI/azure-sdk-for-python) https://github.com/Azure/azure-sdk-for-python/pull/1752
